test/cppsuite/tests/bounded_cursor_perf.cpp

```diff
@@ -81,13 +81,12 @@ class bounded_cursor_perf : public test {
           conn->compile_configuration(conn, "WT_CURSOR.bound", "bound=upper", &compiled_ptr_upper));
 
         /* Initialize the different timers for each function. */
-        execution_timer bounded_next("bounded_next_ticks", test::_args.test_name);
-        execution_timer bounded_prev("bounded_prev_ticks", test::_args.test_name);
-        execution_timer default_next("default_next_ticks", test::_args.test_name);
-        execution_timer default_prev("default_prev_ticks", test::_args.test_name);
-        execution_timer regular_config_timer(
-          "set_bounds non-compiled ticks", test::_args.test_name);
-        execution_timer compiled_config_timer("set_bounds compiled ticks", test::_args.test_name);
+        execution_timer bounded_next("bounded_next", test::_args.test_name);
```

Comment:
Are these counters all per-read?

Reply:
Yeah they're per next call, if that's what you mean?

Reply:
Yep, that's what I meant. Depending on how short-lived these calls are, it might be worth doing the reads in a loop, measuring that time, and dividing - from memory, Linux implements these perf stats using `rdpmc` (or whatever the Arm equivalent is), which is serialising. So for "short-lived" functions you may accidentally be profiling serialisation cost instead of `cursor->read` cost.

Reply:
Does the serialization only occur if the perf counter is enabled? Because I could disable it for the bounded cursor test at it didn't used to count instructions. Though I suppose your comment would apply for other places that we loop in.

Reply:
It would occur whenever Linux uses `rdpmc`, which I would assume is only if the counter is enabled. But yeah, anywhere that's a "small enough" section of code measured would be impacted by this.

If you want to make a ticket and think about this later, that works for me too.

Reply:
I'll experiment with this before merging, the less changes I make after merge the less introduced noise into our produced data.

Reply:
@korteland a bit of experimentation confirms your theory the times with counter enabled are about a factor of 10 worse. I'll have to rethink my implementation, thanks for the heads up.

Reply:
Ouch - that's really unfortunate. I'd be happy to help with this.

Reply:
I ended up rejigging the whole thing: [294b762](https://github.com/wiredtiger/wiredtiger/pull/10602/commits/294b7622cceefe23011363930c8a961d620ebd45)[294b762](https://github.com/wiredtiger/wiredtiger/pull/10602/commits/294b7622cceefe23011363930c8a961d620ebd45)

Reply:
Wow, that's quite the refactor - the overall shape looks good, thanks for taking that on. I think the measurements are in a much better place after splitting the timing/instruction-counting benchmarks.

I didn't review the gory details of that change since it looked like a rearrangement rather than a rewrite. I can go over the details again if you'd like.